pkg/util/util.go

```diff
@@ -247,7 +259,26 @@ func GetDNS1123Name(str string) string {
 		":", "-",
 		"--", "-",
 	)
-	return strings.TrimSpace(replacer.Replace(strings.ToLower(str)))
+	withReplacedChars := replacer.Replace(strings.ToLower(str))
+	return removeNonAlphaSuffix(removeNonAlphaPrefix(withReplacedChars))
```

Comment:
This failed for a name like __#nodejs-ex#@@-3.5
```
[mrinaldas@localhost odo]$ ./odo create nodejs --local /home/mrinaldas/__#nodejs-ex#@@-3.5/
 ✓  Checking component
 ✓  Checking component version
nodejs-ex#@@-nodejs-jqby is not a valid name:  a DNS-1123 label must consist of lower case alphanumeric characters or '-', and must start and end with an alphanumeric character (e.g. 'my-name',  or '123-abc', regex used for validation is '[a-z0-9]([-a-z0-9]*[a-z0-9])?')
```
I think it might be good idea to remove all the non-alpha characters not sure about it though, WDYT?

Reply:
Yeah, fixed